### DISCUSSION TITLE
Authentication in an iframe

### DISCUSSION THREAD

**Discussion by giuseppemag** [zitadel/zitadel#6911] [Q&A]
Dear all,

I am currently building a .Net (Core) and React application. I would like to automatically show an iframe in a sort of overlay/modal whenever the user is detected to be logged out.

If I just open a new tab with the authentication page, the whole redirect flow works fine. If, on the other hand, I show the same page in an iframe, during the login flow Zitadel prompts a CSRF error and does not allow logging in.

I already enabled Zitadel to run in an iframe, and indeed I do see the first page of the login flow in Zitadel so something is working, but also clearly I am missing something. Maybe I need to set a header or something?

Advice would be greatly appreciated <3

**Comment by livio-a** [comment]
Hey @giuseppemag 

I think you're also part of the Discord thread regarding this topic. As just mentioned there, we've created a fix and release it with 2.41.1. The deployment to zitadel.cloud has just been finished as well.
